fedimint-rocksdb/src/lib.rs

```diff
@@ -51,6 +51,27 @@ impl From<RocksDb> for rocksdb::OptimisticTransactionDB {
     }
 }
 
+// When finding by prefix iterating in Reverse order, we need to start from
+// "prefix+1" instead of "prefix", using lexicographic ordering. See the tests
+// below
+fn next_prefix(prefix: &[u8]) -> Result<Vec<u8>> {
+    let mut next_prefix = prefix.to_vec();
+    let mut increased = false;
+    for i in (0..next_prefix.len()).rev() {
+        if next_prefix[i] < 0xff {
+            next_prefix[i] += 1;
+            increased = true;
+            break;
+        } else {
+            next_prefix[i] = 0;
+        }
+    }
+    if !increased {
+        bail!("Prefix is already the max one: {prefix:?}")
```

Comment:
I don't think we should panic here. Seems like handling `0xff...` should be taken care of. Return an `Option` maybe?

Reply:
It isn't panic, it's an error. In fact it could be a panic because it's a bug (an invalid prefix has been created).

But if we return an option the caller should do the same, IMHO.

Reply:
Huh, why if `0xff` an invalid prefix? We might declare it as such because it's convenient, but it's not obviously true to me. 

Correct me if I'm wrong, but in that case we might want to return the empty prefix `vec![]` so that we start iterating from the absolutely last element when iterating in reverse order. Naturally the `0xff` prefix should come first then.